### PR TITLE
TimeInput with Actual Data

### PR DESCRIPTION
**Pull request by praspra2**
# Related Issue

A Pull Request should close a **single** issue; multiple issues can be closed when the issues are **small and related**, but that should be an exception not the rule. Please keep Pull Requests small and targeted; large 'code drops' with dozens of files will be closed and asked to split into reviewable pieces. Reviews that need to be large due to dependencies will be
reviewed on a case-by-case basis.

Please use one of the well-known [github fixes keywords](https://help.github.com/en/articles/closing-issues-using-keywords) to reference
the issue fixed with this PR (eg Fixes #<github issue number>). If an issue doesn't yet exist please create one to aid
in issue tracking.

**NOTE**: For multiple issues resolved by this PR use the corresponding keywords **every time** in a comma-delimited list per the reference
page above.

# Description

For all Pull Requests, please describe how the issue was fixed or how the feature was implemented from a summary level. This information will be used to help provide context to the reviewers of the pull request and should be additive to the issue being closed.

- Implemented Input Time element with actual data using the model based approach.
- https://app.vidcast.io/share/c783a65f-79f4-4af0-a5a3-6a3950fe88a3

image
  
image

# Sample Card

If appropriate, please include a link to a card in one of the samples directories that can be used to validate this change. This can be an existing card or a card added with this PR.

# How Verified

How you verified the fix, including one or all of the following:
1. New unit tests that were added if any. If none were added please add a quick line explaining why not.
2. Existing relevant unit/regression tests that you ran
3. Manual scenario verification if any; ***Do include .gif's or screenshots of the testing you performed here if you think that it
will aid in code reviews or corresponding fixes on other platforms for eg.***

### CODE REVIEW

File: source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.cpp [hunk 1]
```diff
@@ -5,8 +5,54 @@
 #include "MarkDownParser.h"
 
 TimeInputModel::TimeInputModel(std::shared_ptr<AdaptiveCards::TimeInput> input, QObject* parent) :
-	QObject(parent)
+    QObject(parent),
+    mTimeInputConfig(AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getCardConfig()->getInputTimeConfig())
 {
+    const auto hostConfig = AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getHostConfig();
+    const auto rendererConfig = AdaptiveCardQmlEngine::AdaptiveCardContext::getInstance().getCardConfig();
+
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.cpp:13]: Check this and remove the unused.

Comment (praspra2) [reply source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.cpp:12]: Done

File: source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.h [hunk 2]
```diff
@@ -11,9 +12,74 @@ class TimeInputModel : public QObject
...
+    QString mCurrHour = "-1";
+    QString mCurrMinute = "-1";
+
+    QRegExp mRegex;
+
+    bool mIsVisible{false};
+    bool mIsRequired{false};
+    bool mIs12hour;
+    bool mValidationRequired;
+};
```
Comment (soumykun) [inline source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.h:85]: The remaining 2 are not initialized ?

**Review by soumykun — APPROVED** [review]
Will address the initialization comment in a separate PR.

File: source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.h (hunk 2 above)
Comment (soumykun) [reply source/qml_v2/AdaptiveCardQmlEngine/models/TimeInputModel.h:85]: Will address in a separate PR following this.

**Review by ramkrishnajana — APPROVED** [review]

**Pull request merged**